Documentation/Reference.md

```diff
@@ -61,51 +61,117 @@ Where `foo` is an executable and `bar.a` a static library.
+
+```swift
+import PackageDescription
+
+let package = Package(
+    name: "FooBar"
+)
+```
+
+### targets
+
+Targets property is used when you have more than one module in your package and want to declare inter-dependency between them.
+
+Note: It is also possible to declare target dependencies between a test and regular module.
 
 ```swift
 import PackageDescription
 
 let package = Package(
     name: "Hello",
-    dependencies: [
-        .Package(url: "ssh://git@example.com/Greeter.git", versions: Version(1,0,0)..<Version(2,0,0)),
+    targets: [
+        Target(name: "TestSupport"),
+        Target(name: "Bar", dependencies: ["Foo"]),
+        Target(name: "BarTests", dependencies: ["Bar", "TestSupport"]),
     ]
 )
 ```
 
-A `Package.swift` file is a Swift file that declaratively configures a Package using types defined in the `PackageDescription` module. This manifest declares a dependency on an external package: `Greeter`.
+Here `Foo`, `Bar` and `TestSupport` are modules present under `Sources/` directory and `BarTests` is a test module under `Tests/` directory.
+`Bar` module can import `Foo` module and `BarTests` module can import `Foo`, `Bar` and `TestSupport` modules.
 
-If your package contains multiple targets that depend on each other you will need to specify their interdependencies. Here is an example:
+### dependencies
+
+This is the list of packages that the current package depends on. You can specify url (or local path) to any valid swift package placed under
+version control which is tagged appropriately.
 
 ```swift
 import PackageDescription
 
 let package = Package(
     name: "Example",
-    targets: [
-        Target(
-            name: "top",
-            dependencies: [.Target(name: "bottom")]),
-        Target(
-            name: "bottom")
+    dependencies: [
+        .Package(url: "ssh://git@example.com/Greeter.git", versions: Version(1,0,0)..<Version(2,0,0)),
+        .Package(url: "../StringExtensions", majorVersion: 1),
+        .Package(url: "https://github.com/MyAwesomePackage", majorVersion: 1, minor: 4),
     ]
 )
 ```
 
-The targets are named how your subdirectories are named.
+### exclude
+
+Use this property to exclude files and directories from the package sources.
 
-If you want to exclude some files and folders from Package, you can simple list them in the `exclude`. Every item specifies a relative folder path from the Root folder of the package
+Every item specifies a relative path from the root of the package.
 
 ```swift
 let package = Package(
-    name: "Example",
-    exclude: ["tools", "docs", "Sources/libA/images"]
+    name: "Foo",
+    exclude: ["Sources/Fixtures", "Sources/readme.md", "Tests/FooTests/images"]
 )
 ```
 
+### pkgconfig
+
+This property should only be used for System Module Packages. It defines the name of the pkg-config (.pc) file
```

Comment:
BTW, is it an error not to use this for system module packages? We should make it so, if not.